istioctl/pkg/multicluster/remote_secret.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2019 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicluster
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/runtime/serializer/versioning"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/tools/clientcmd/api/latest"
+
+	"istio.io/istio/pkg/kube"
+)
+
+var (
+	Codec  runtime.Codec
+	Scheme *runtime.Scheme
+)
+
+func init() {
+	Scheme = runtime.NewScheme()
+	utilruntime.Must(v1.AddToScheme(Scheme))
+	yamlSerializer := json.NewYAMLSerializer(json.DefaultMetaFactory, Scheme, Scheme)
+	Codec = versioning.NewDefaultingCodecForScheme(
+		Scheme,
+		yamlSerializer,
+		yamlSerializer,
+		v1.SchemeGroupVersion,
+		runtime.InternalGroupVersioner,
+	)
+}
+
+type options struct {
+	secretPrefix       string
+	serviceAccountName string
+	secretLabels       map[string]string
+
+	// inherited from root command
+	namespace  string
+	kubeconfig string
+
+	args []string
+}
+
+var (
+	defaultSecretPrefix = "istio-pilot-remote-secret-"
+	defaultSecretlabels = map[string]string{
+		"istio/multiCluster":            "true", // legacy label
+		"istio.io/remote-multi-cluster": "true",
+	}
+	defaultServiceAccountName = "istio-pilot-service-account"
+)
+
+// NewCreatePilotRemoteSecretCommand creates a new command for joining two clusters togeather in a multi-cluster mesh.
+func NewCreatePilotRemoteSecretCommand(kubeconfig, namespace *string) *cobra.Command {
+	o := options{
+		secretPrefix:       defaultSecretPrefix,
+		secretLabels:       defaultSecretlabels,
+		serviceAccountName: defaultServiceAccountName,
+	}
+
+	c := &cobra.Command{
+		Use:   "create-pilot-remote-secrets [list of remote clusters]",
```

Comment:
At least one remote cluster must be specified so instead of `[list of remote clusters]` perhaps something like `cluster1 [clusterN]*`? Add `[--service-account-name=<sa>]`?

Reply:
This has been changed to exactly one cluster: the current context.